spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesDiscoveryProperties.java

```diff
@@ -175,100 +177,25 @@ public String toString() {
+	 * @param annotationsPrefix prefix for the annotations
+	 * @param addPorts include ports as metadata
+	 * @param portsPrefix prefix for the ports, by default it is "port."
 	 */
-	public static class Metadata {
-
-		/**
-		 * When set, the Kubernetes labels of the services will be included as metadata of
-		 * the returned ServiceInstance.
-		 */
-		private boolean addLabels = true;
-
-		/**
-		 * When addLabels is set, then this will be used as a prefix to the key names in
-		 * the metadata map.
-		 */
-		private String labelsPrefix;
-
-		/**
-		 * When set, the Kubernetes annotations of the services will be included as
-		 * metadata of the returned ServiceInstance.
-		 */
-		private boolean addAnnotations = true;
-
-		/**
-		 * When addAnnotations is set, then this will be used as a prefix to the key names
-		 * in the metadata map.
-		 */
-		private String annotationsPrefix;
-
-		/**
-		 * When set, any named Kubernetes service ports will be included as metadata of
-		 * the returned ServiceInstance.
-		 */
-		private boolean addPorts = true;
-
-		/**
-		 * When addPorts is set, then this will be used as a prefix to the key names in
-		 * the metadata map.
-		 */
-		private String portsPrefix = "port.";
-
-		public boolean isAddLabels() {
-			return this.addLabels;
-		}
-
-		public void setAddLabels(boolean addLabels) {
-			this.addLabels = addLabels;
-		}
-
-		public String getLabelsPrefix() {
-			return this.labelsPrefix;
-		}
-
-		public void setLabelsPrefix(String labelsPrefix) {
-			this.labelsPrefix = labelsPrefix;
-		}
-
-		public boolean isAddAnnotations() {
-			return this.addAnnotations;
-		}
-
-		public void setAddAnnotations(boolean addAnnotations) {
-			this.addAnnotations = addAnnotations;
-		}
-
-		public String getAnnotationsPrefix() {
-			return this.annotationsPrefix;
-		}
-
-		public void setAnnotationsPrefix(String annotationsPrefix) {
-			this.annotationsPrefix = annotationsPrefix;
-		}
+	public record Metadata(@DefaultValue("true") boolean addLabels, String labelsPrefix,
+			@DefaultValue("true") boolean addAnnotations, String annotationsPrefix,
+			@DefaultValue("true") boolean addPorts, @DefaultValue("port.") String portsPrefix) {
 
-		public boolean isAddPorts() {
-			return this.addPorts;
-		}
-
-		public void setAddPorts(boolean addPorts) {
-			this.addPorts = addPorts;
-		}
-
-		public String getPortsPrefix() {
-			return this.portsPrefix;
-		}
+		@ConstructorBinding
+		public Metadata {
 
-		public void setPortsPrefix(String portsPrefix) {
-			this.portsPrefix = portsPrefix;
 		}
 
-		@Override
-		public String toString() {
-			return new ToStringCreator(this).append("addLabels", this.addLabels)
-					.append("labelsPrefix", this.labelsPrefix).append("addAnnotations", this.addAnnotations)
-					.append("annotationsPrefix", this.annotationsPrefix).append("addPorts", this.addPorts)
-					.append("portsPrefix", this.portsPrefix).toString();
+		// needed in order to get the defaults for some fields
```

Comment:
we need a no-arg constructor because the class using this (the outer class) has a calls as : `new Metadata()` and I wanted to keep it, as it is easy to follow.